### ISSUE TITLE
How to run UDP instead of TCP, str2str -> rtkrcv

### ISSUE THREAD

**Issue by olofcarlsson** [tomojitakasu/RTKLIB#403]
Hi, I want to transmit correction data from a base station to several rovers(>10) over a radio-link. For this application it would be nice to run the base station application str2str in UDP-mode instead of TCP. Is there any work done on this? Or any idea on how this could be implemented?

**Comment by joriatyBen** [comment]
Could be possible with netcat

olofcarlsson <notifications@github.com> schrieb am Fr., 20. Apr. 2018,
07:13:

**Comment by nprobert** [comment]
Yes, we would like this feature for str2str to also output RTCM data via UDP to a Roadside Unit (RSU) for V2X communcations.

**Comment by DavidKelleySCSC** [comment]
Neal, given your skill in programming, it would be faster to to just build up an NTRIP client and then have that do a UDP forward to the RSU with any additional value add that made sense for the RSU interface.  Also, NTRIP Rev2 has provisions for UDP use already done, but they are not implemented in RTKLIB.  To the original post from olofcarlsson, there are a number of radio makers who can accept TCP/IP and broadcast that.  You simply take str2str and point it at the radio. Of course in both of these you are limited to the one data stream.

**Comment by nprobert** [comment]
That's what I'm thinking now.  The Linux client will use RTKLIB for NTRIP and do the J2735 RTCM message encoding (using asn1c) then forward over UDP.  We have a LearRSU which will forward data OTA from UDP, but it doesn't have RTCM support or do J2735 encoding.

**Comment by DavidKelleySCSC** [comment]
I designed SAE J2735 so you (as an implementer) would not need to understand RTCM content, just to convey the payloads along to the GNSS in the OBU.  If I am missing something there, feel free to PM me.
